cypress/integration/mocked/okta_sign_in.spec.ts

```diff
@@ -0,0 +1,175 @@
+      });
+      cy.get('[data-cy=sign-in-button]').click();
+      cy.contains('There was a problem signing in, please try again.');
+    });
+
+    it('loads the redirectUrl upon successful authentication', function () {
+      cy.visit(
+        '/signin?returnUrl=https%3A%2F%2Fwww.theguardian.com%2Fabout&useOkta=true',
+      );
+      cy.get('input[name="email"]').type('example@example.com');
+      cy.get('input[name="password"]').type('password');
+      cy.mockNext(200, {
+        expiresAt: '3000-01-01T00:00:00.000Z',
+        status: 'SUCCESS',
+        sessionToken: 'some-session-token',
+        _embedded: {
+          user: {
+            id: 'okta-id',
+            passwordChanged: '2020-01-01T00:00:00.000Z',
+            profile: {
+              login: 'test.man@example.com',
+              firstName: 'Test',
+              lastName: 'Man',
+              locale: 'en_GB',
+              timeZone: 'Europe/London',
+            },
+          },
+        },
+      });
+
+      // we can't actually check the authorization code flow
+      // so intercept the request and redirect to the guardian about page
+      cy.intercept(
+        `http://localhost:9000/oauth2/${Cypress.env(
+          'OKTA_CUSTOM_OAUTH_SERVER',
+        )}/v1/authorize*`,
+        (req) => {
+          req.redirect(defaultReturnUrl);
+        },
+      ).as('authRedirect');
+
+      cy.get('[data-cy=sign-in-button]').click();
+      cy.waitFor('@authRedirect').then(() => {
+        cy.url().contains('https://theguardian.com/about');
+      });
+    });
+
+    it('redirects to the default url if no redirectUrl given', function () {
+      cy.visit('/signin?useOkta=true');
+      cy.get('input[name="email"]').type('example@example.com');
+      cy.get('input[name="password"]').type('password');
+      cy.mockNext(200, {
+        expiresAt: '3000-01-01T00:00:00.000Z',
+        status: 'SUCCESS',
+        sessionToken: 'some-session-token',
+        _embedded: {
+          user: {
+            id: 'okta-id',
+            passwordChanged: '2020-01-01T00:00:00.000Z',
+            profile: {
+              login: 'test.man@example.com',
+              firstName: 'Test',
+              lastName: 'Man',
+              locale: 'en_GB',
+              timeZone: 'Europe/London',
+            },
+          },
+        },
+      });
+
+      cy.get('[data-cy=sign-in-button]').click();
+      // we can't actually check the authorization code flow
+      // so intercept the request and redirect to the guardian about page
+      cy.intercept(
+        `http://localhost:9000/oauth2/${Cypress.env(
+          'OKTA_CUSTOM_OAUTH_SERVER',
+        )}/v1/authorize*`,
+        (req) => {
+          req.redirect(defaultReturnUrl);
+        },
+      ).as('authRedirect');
+
+      cy.get('[data-cy=sign-in-button]').click();
+      cy.waitFor('@authRedirect').then(() => {
+        cy.url().contains('https://theguardian.com/about');
+      });
+    });
+  });
+});
```

Comment:
Note to self as much as anything, might make sense to having a equivalent to the tests that start with

```js
context('Redirect to post sign-in prompt', () => {
```

in the IDAPI sign in tests, now these aren't skipped.